### PR TITLE
store/tikv: move binlog logic to txn_driver

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
store/tikv depends on binlog utilities.
Part of https://github.com/pingcap/tidb/issues/22513

### What is changed and how it works?
What's Changed:
- move concret binlog logic to txn driver and inject it into 2pc

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -1456,7 +1454,7 @@ func (c *twoPhaseCommitter) calculateMaxCommitTS(ctx context.Context) error {
 }
 
 func (c *twoPhaseCommitter) shouldWriteBinlog() bool {
-	return c.txn.us.GetOption(kv.BinlogInfo) != nil
+	return c.txn.us.GetOption(kv.BinlogInfo) != nil && c.binlog != nil
```
Comment (AndreMouche) [inline store/tikv/2pc.go:1457]: Maybe we could remove the option `kv.BinlogInfo` in the future, when the driver set the `kv.BinlogInfo` option, we init `c.binlog`

Comment (disksing) [reply store/tikv/2pc.go:1457]: good idea

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**